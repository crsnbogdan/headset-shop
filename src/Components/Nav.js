import AddIcon from '@mui/icons-material/Add';
import CloseIcon from '@mui/icons-material/Close';
import DeleteIcon from '@mui/icons-material/Delete';
import MenuIcon from '@mui/icons-material/Menu';
import RemoveIcon from '@mui/icons-material/Remove';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
import { AnimatePresence, motion } from 'framer-motion';
import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import uniqid from 'uniqid';
import '../Styles/Nav.css';

const Nav = (props) => {
  const [mobileNavVisible, setMobileNavVisible] = useState(false);

  let cartProducts = props.cart.map((cartItem) => {
    return (
      <motion.div
        initial={{ opacity: 0 }}
        animate={{ opacity: 1 }}
        transition={{ duration: 0.25 }}
        exit={{ opacity: 0, display: 'none' }}
        className="nav__cartitem grid p-2 bg-white"
        key={uniqid()}
      >
        <img className="cartitem__img" src={cartItem.product.image} alt="" />
        <div className="cartitem__info flex flex-col justify-between text-sm text-black">
          <p className="text-black font-semibold cartitem_itemtotal">
            <span className="font-normal">Total: </span>
            {(
              cartItem.qty *
              (cartItem.product.salePrice || cartItem.product.regularPrice)
            ).toFixed(2)}
            $
          </p>
          <p className="text-sm">
            {cartItem.product.name.substring(0, 40).trim() + '(...)'}
          </p>
          <p className="cartitem__qtyval">Quantity: {cartItem.qty}</p>
        </div>
        <div className="cartitem__actions justify-between items-center flex flex-col">
          <DeleteIcon
            className="cartitem__delete"
            onClick={() => {
              props.setCartQuantity(props.cartQuantity - cartItem.qty);

              let productInCartIndex = props.cart.findIndex(
                (item) => cartItem.sku === item.sku
              );
              let cachedCart = props.cart;
              cachedCart.splice(productInCartIndex, 1);
              props.setCart([...cachedCart]);
            }}
          />
          <div className="cartitem__qtymodifiers flex items-center ">
            <AddIcon
              className="cartitem__qtymodifier add"
              onClick={() => {
                props.setCartQuantity(props.cartQuantity + 1);
                let productInCartIndex = props.cart.findIndex(
                  (item) => cartItem.sku === item.sku
                );
                let cachedCart = props.cart;
                cachedCart[productInCartIndex].qty += 1;
                props.setCart([...cachedCart]);
              }}
            />
            <p className="text-lg text-black font-semibold py-1 px-3 bg-slate-100">
              {cartItem.qty}
            </p>
            <RemoveIcon
              className="cartitem__qtymodifier subtract"
              onClick={() => {
                props.setCartQuantity(props.cartQuantity - 1);
                let productInCartIndex = props.cart.findIndex(
                  (item) => cartItem.sku === item.sku
                );
                let cachedCart = props.cart;
                cachedCart[productInCartIndex].qty -= 1;
                if (cachedCart[productInCartIndex].qty === 0) {
                  cachedCart.splice(productInCartIndex, 1);
                  if (cachedCart.length === 0) {
                    props.setCartVisible(false);
                  }
                  return props.setCart([...cachedCart]);
                }
                props.setCart([...cachedCart]);
              }}
            />
          </div>
        </div>
      </motion.div>
    );
  });

  return (
    <div className="nav__container">
      <nav className="nav items-center relative text-white flex px-28 py-6 justify-between">
        <h3 className="text-2xl text-yellow-300 font-semibold">
          KB<span className="text-white font-light text-lg">.shop</span>
        </h3>
        <MenuIcon
          className="nav__sidebartoggle"
          fontSize="large"
          onClick={() => {
            setMobileNavVisible(true);
            document
              .querySelector('.page__navoverlay')
              .classList.toggle('visible');
          }}
        />
        <motion.div
          initial={{ right: -300 }}
          animate={{
            right: mobileNavVisible ? 0 : -300,
          }}
          transition={{ duration: 0.5 }}
          className="nav__links w-3/12 text-lg flex items-center justify-between"
        >
          <CloseIcon
            className="nav__close"
            fontSize="large"
            onClick={() => {
              setMobileNavVisible(false);
              document
                .querySelector('.page__navoverlay')
                .classList.toggle('visible');
            }}
          />
          <NavLink
            to="/"
            end
            className="nav__link cart"
            onClick={() => {
              setMobileNavVisible(false);
              document
                .querySelector('.page__navoverlay')
                .classList.toggle('visible');
            }}
          >
            home
          </NavLink>
          <NavLink
            to="shop"
            end
            className="nav__link"
            onClick={() => {
              setMobileNavVisible(false);
              document
                .querySelector('.page__navoverlay')
                .classList.toggle('visible');
            }}
          >
            shop
          </NavLink>
          <NavLink
            to="cart"
            className="nav__link relative"
            onClick={() => {
              setMobileNavVisible(false);
              document
                .querySelector('.page__navoverlay')
                .classList.toggle('visible');
            }}
          >
            {props.cartQuantity > 0 ? (
              <motion.div
                initial={{ scale: 0.5 }}
                animate={{ scale: 1 }}
                transition={{ duration: 0.25 }}
                className="nav__qty flex justify-center items-center color-white absolute text-xs font-medium rounded-full"
              >
                <p className="text-white">{props.cartQuantity}</p>
              </motion.div>
            ) : null}
            <ShoppingCartIcon
              fontSize="medium"
              sx={{ padding: '10px', boxSizing: 'content-box' }}
              onMouseEnter={() => props.setCartVisible(true)}
              onMouseLeave={() => props.setCartVisible(false)}
            />
          </NavLink>
        </motion.div>
        <AnimatePresence>
          {props.cartVisible && (
            <motion.div
              initial={{ minHeight: '0px' }}
              animate={{ minHeight: '100px' }}
              exit={{ minHeight: '0px' }}
              transition={{ duration: 0.5, delayChildren: 1 }}
              onMouseEnter={() => props.setCartVisible(true)}
              onMouseLeave={() => props.setCartVisible(false)}
              className="nav__cartcontainer p-2 bg-slate-100 absolute"
            >
              {props.cart.length === 0 && props.cartVisible && (
                <motion.p
                  initial={{ display: 'none' }}
                  animate={{ display: 'block' }}
                  exit={{ display: 'none' }}
                  transition={{ delay: 0.2 }}
                  className="nav__cartemptytext text-slate-400"
                >
                  The cart is empty
                </motion.p>
              )}
              {props.cart && props.cartVisible && cartProducts}
              {props.cart.length > 0 && props.cartVisible && (
                <motion.p
                  exit={{ display: 'none' }}
                  className="text-black nav__carttotal"
                >
                  Total:{' '}
                  <span className="font-semibold">
                    {props.cart
                      .reduce(
                        (total, product) =>
                          total +
                          product.qty *
                            (product.product.salePrice ||
                              product.product.salePrice),
                        0
                      )
                      .toFixed(2)}
                    $
                  </span>
                </motion.p>
              )}
            </motion.div>
          )}
        </AnimatePresence>
      </nav>
    </div>
  );
};

export default Nav;
