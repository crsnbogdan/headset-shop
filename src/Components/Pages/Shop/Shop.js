import { AnimatePresence, motion } from 'framer-motion';
import React, { useState } from 'react';
import Typewriter from 'typewriter-effect';
import '../../../Styles/Shop.css';
import ShopProducts from './ShopProducts';

import FormControl from '@mui/material/FormControl';
import InputLabel from '@mui/material/InputLabel';
import MenuItem from '@mui/material/MenuItem';
import Select from '@mui/material/Select';

const Shop = (props) => {
  const [cartPopup, setCartPopup] = useState(false);
  const [sortType, setSortType] = useState('reviews');

  return (
    <div className="shop bg-blue-300">
      <div className="page__navoverlay" />
      <div className="shop__hero relative flex items-center justify-center bg-slate-200">
        <motion.h1
          initial={{ y: '-20px', opacity: 0 }}
          animate={{ y: 0, opacity: 1 }}
          transition={{ duration: 0.25 }}
          className="text-white text-6xl font-bold"
        >
          <Typewriter
            options={{
              autoStart: true,
              delay: 'natural',
              deleteSpeed: Infinity, // Infinity set to disable
              // the deletion of the string
              pauseFor: 1000,
              strings: ['shop'],
              wrapperClassName:
                'shop__typewriter text-white text-5xl font-bold text-white',
              cursorClassName:
                'shop__cursor text-5xl font-light text-yellow-300',
            }}
          />
        </motion.h1>
      </div>
      <div className="shop__maincontainer py-12 px-24  bg-slate-100 h-full">
        <AnimatePresence>
          {cartPopup && (
            <motion.div
              initial={{ y: 100, opacity: 0.5 }}
              animate={{ y: 0, opacity: 1 }}
              transition={{ duration: 0.5 }}
              exit={{ y: 100, opacity: 0.5 }}
              className="shop_addedpopup flex justify-center items-center absolute bg-white"
            >
              <p className="text-slate-400 py-3">Added to cart.</p>
            </motion.div>
          )}
        </AnimatePresence>
        <motion.div
          className="flex justify-between items-center mb-4"
          initial={{ y: '-10px', opacity: 0 }}
          animate={{ y: 0, opacity: 1 }}
          transition={{ duration: 1 }}
        >
          <h2 className="text-2xl">Products:</h2>
          <FormControl sx={{ width: '160px' }}>
            <InputLabel id="demo-simple-select-label">Sort by</InputLabel>
            <Select
              value={sortType}
              label="sortType"
              onChange={(e) => setSortType(e.target.value)}
            >
              <MenuItem value="reviews">Review count</MenuItem>
              <MenuItem value="rating">Rating</MenuItem>
              <MenuItem value="priceAsc">Price ( asc. )</MenuItem>
              <MenuItem value="priceDesc">Price ( des. )</MenuItem>
            </Select>
          </FormControl>
        </motion.div>
        <ShopProducts
          productsArr={props.productsArr}
          updateCartQuantity={props.updateCartQuantity}
          cart={props.cart}
          sortType={sortType}
          setCartPopup={setCartPopup}
          cartPopup={cartPopup}
          setCart={props.setCart}
        />
      </div>
    </div>
  );
};

export default Shop;
