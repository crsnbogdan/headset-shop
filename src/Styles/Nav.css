nav {
    background: linear-gradient(180deg, rgb(13, 23, 45) 15%, rgb(16, 9, 61) 100%);
    height: 10vh;
    border-bottom: 2px solid rgb(253,224, 71);
}

.nav__link {
    color: white;
}
.nav__link span {
    color: #fde047;
}
.nav__link.active {
    color: #fde047;
}

#cart__underline {
    width: 0%;
    height: 2px;
    background-color: red;
}

.nav__link {
    border-bottom: 0px solid #fde047;
    transition: 0.5s ease-in;
}
.nav__link.active {
    border-bottom: 2px solid #fde047;
}

.nav__qty {
 pointer-events: none;
 background-color: red;
 width: 20px;
 height: 20px;
 top: -6px;
 right: -4px;
}

@media screen and (max-width: 1024px) {
    nav .nav__links  {
        width: 50%!important;
    }
}
@media screen and (max-width: 550px) {
    .nav {
        padding: 20px 40px!important;
    }
    .nav .nav__links  {
        width: 50%!important;
    }
}