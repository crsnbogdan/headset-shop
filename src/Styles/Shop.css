.shop {
    min-height: 75vh;
}
.page__navoverlay {
    display: none;
}
.page__navoverlay.visible {
    display: block;
    z-index: 5; 
    width: 100vw;
    height: 100%;
    position: fixed;
    top: 0;
    background-color: #00000086;
}

.shop__hero {
    height: 150px;
    margin-top: 10vh;
    background-position: center;
    background-size: 100% auto;
    background-image: url('../Media/shop-bg.jpg');
}

.shop__hero h1 {
    text-shadow: 0 0 10px rgb(0, 0, 0);
    margin-top: -15px;
    
}


.shop__maincontainer {
    min-height: 75vh;
    max-width: 100%;
}

.shop_addedpopup {
    width: 300px;
    right: 6rem;
    bottom: 30px;
    -webkit-box-shadow: 0px 0px 15px 0px #00000035; 
    box-shadow: 0px 0px 15px 0px #00000035;
    z-index: 5;
}

.shop__productscontainer {
    width: 100%;
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    gap: 20px;
}

.shop__productcontainer {
    width: 100%;
    height: 100%;
    place-self: center;
}
.nav__cartemptytext {
    place-self: center;
}

.nav__carttotal {
    place-self: center;
    grid-column: 1 / -1;
}

.shop__product {
    align-self: right;
    width: 100%;
    background-color: white;
    padding: 10px;
    position: relative;
}

.shop__productimg {
    width: 100%;
    height: 175px;
    background-size: 100% auto ;
    background-position: center;
    background-repeat: no-repeat;
}

.product__price--sale {
    color: red;
}

.product__price--sale:after {
    content: 'SALE';
    font-weight: 600;
    color: black;
    background-color: #fde047; 
    padding: 2px 8px;
    border-radius: 5px;
    right: 10px;
    top: 10px;
    position: absolute;
}

.product__addbtn {
    overflow: hidden;
    color: white;
    background-image: linear-gradient(to right, rgb(29, 78, 216), rgb(126, 34, 206) 75%);
}

.addbtn__icon {
    position: absolute;
    width: 48px!important;
    height: 32px!important;
    padding: 0.4rem!important;
    top: 0;
    left: -5px;
    border-radius: 5px;
    border-top-right-radius: 0px;
    border-bottom-right-radius: 200px;
    background-color: #5184fa;
}

#root > div > div > div.shop__hero.relative.flex.items-center.justify-center.bg-slate-200 > h1 > div > span.shop__cursor.text-5xl.font-light.text-yellow-300 { 
    animation: cursor__blink 1.5s linear infinite;
    animation-delay: 1.5s;
}

@keyframes cursor__blink {
    0% {
        opacity: 0;
    }

    50% {
        opacity: 1;
    }

    60% {
        opacity: 0;
    }

    100% {
        opacity: 0;
    }
}

@media screen and (max-width: 1400px) {
    .shop__productscontainer {
        grid-template-columns: repeat(3, 1fr);
    }

    .shop__productcontainer {
        width: 100%;
    }

    .shop__product {
        width: 100%;
        height: 100%;
    }
}

@media screen and (max-width: 800px) {
    .shop__maincontainer {
        padding: 1.5rem!important;
    }
    .shop_addedpopup{
        right: 25%;
        width: 50vw;
        transform: translateX(50%);
    }

    .shop__productscontainer {
        grid-template-columns: repeat(2, 1fr);
    }
}

@media screen and (max-width: 500px) {
    .shop__productscontainer {
        grid-template-columns: repeat(1, 1fr);
    }
}