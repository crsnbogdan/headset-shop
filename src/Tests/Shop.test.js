import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Nav from '../Components/Nav';
import Shop from '../Components/Pages/Shop/Shop';

let productsArr = [
  {
    name: 'Product 1',
    customerReviewAverage: 4.7,
    customerReviewCount: 2267,
    regularPrice: 99.99,
    salePrice: 89.99,
    longDescription: 'This is a test',
    image:
      'https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/5053/5053501_sd.jpg',
    sku: 1,
  },
  {
    name: 'Product 2',
    customerReviewAverage: 4.6,
    customerReviewCount: 1437,
    regularPrice: 149.99,
    salePrice: 99.99,
    longDescription: 'This is a test',
    image:
      'https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/6298/6298657_sd.jpg',
    sku: 2,
  },
  {
    name: 'Product 3',
    customerReviewAverage: 3.7,
    customerReviewCount: 100,
    regularPrice: 95.99,
    salePrice: 79.99,
    longDescription: 'This is a test',
    image:
      'https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/6298/6298657_sd.jpg',
    sku: 3,
  },
];

it('renders corrently and matches snapshot', () => {
  const shopTree = renderer
    .create(
      <Shop productsArr={productsArr} sortType="reviews" showStars={false} />
    )
    .toJSON();
  expect(shopTree).toMatchSnapshot();
});

it('renders no products when props.productsArr is empty', () => {
  let { container } = render(<Shop productsArr={[]} sortType="reviews" />);

  expect(
    [...container.querySelectorAll('.shop__productcontainer')].length
  ).toBe(0);
});

it('renders multiple shop products', () => {
  let { container } = render(
    <Shop productsArr={productsArr} sortType="reviews" />
  );

  expect(
    [...container.querySelectorAll('.shop__productcontainer')].length
  ).toBe(3);
});

it('renders all HTML elements required for product cards', () => {
  let { container } = render(
    <Shop productsArr={productsArr} sortType="reviews" showStars={true} />
  );

  expect(container.querySelector('.shop__productimg')).toBeInTheDocument();
  expect(container.querySelector('.product__price')).toBeInTheDocument();
  expect(container.querySelector('.product__ratingcount')).toBeInTheDocument();
  expect(container.querySelector('.star-ratings')).toBeInTheDocument();
  expect(container.querySelector('.product__addbtn')).toBeInTheDocument();
});

it('calls props.setCart and props.setCartQuantity when the "Add to Cart" button of a product card is clicked', () => {
  const setCartQuantityMock = jest.fn();
  const setCartMock = jest.fn();
  let { container } = render(
    <>
      <BrowserRouter>
        <Nav
          cart={[]}
          setCart={setCartMock}
          setCartQuantity={setCartQuantityMock}
          cartQuantity={0}
        />
      </BrowserRouter>
      <Shop
        setCartQuantity={setCartQuantityMock}
        productsArr={productsArr}
        setCart={setCartMock}
        sortType="reviews"
        cart={[]}
      />
    </>
  );

  let btns = [...container.querySelectorAll('.product__addbtn')];
  btns.forEach((btn) => userEvent.click(btn));
  expect(setCartQuantityMock).toHaveBeenCalledTimes(3);
  expect(setCartMock).toHaveBeenCalledTimes(3);
});

it('increases props.cartQuantity and ".nav__qty" when the "Add to Cart" button of a product card is clicked', () => {
  let cartQty = 0;

  const setCartQuantityMock = () => (cartQty += 1);
  const setCartMock = jest.fn();
  let { container } = render(
    <>
      <BrowserRouter>
        <Nav
          cart={[]}
          setCart={setCartMock}
          setCartQuantity={setCartQuantityMock}
          cartQuantity={cartQty}
        />
      </BrowserRouter>
      <Shop
        setCartQuantity={setCartQuantityMock}
        productsArr={productsArr}
        setCart={setCartMock}
        sortType="reviews"
        cart={[]}
      />
    </>
  );

  expect(container.querySelector('.nav__qty > p')).not.toBeInTheDocument();
  let btns = [...container.querySelectorAll('.product__addbtn')];
  btns.forEach((btn) => userEvent.click(btn));
  expect(cartQty).toBe(3);
  setTimeout(() => {
    expect(container.querySelector('.nav__qty > p').textContent).toBe(4);
  }, 0);
});

it('sorts products based on review count', () => {
  let { container } = render(
    <Shop productsArr={productsArr} sortType="reviews" />
  );

  let productReviewCounts = [
    ...container.querySelectorAll('.product__ratingcount'),
  ];
  productReviewCounts = productReviewCounts.map(
    (product) => product.textContent
  );
  let filteredProductReviewCounts = productReviewCounts.filter((a, b) =>
    a > b ? 1 : -1
  );

  expect(productReviewCounts).toEqual(filteredProductReviewCounts);
});

it('sorts products from highest to lowest price', () => {
  let { container } = render(
    <Shop productsArr={productsArr} sortType="priceDesc" />
  );

  let productPrices = [...container.querySelectorAll('.priceDesc')];
  productPrices = productPrices.map((product) => product.textContent);
  let filteredProductPrices = productPrices.filter((a, b) => (a > b ? 1 : -1));

  expect(productPrices).toEqual(filteredProductPrices);
});

it('sorts products from lowest to highest price', () => {
  let { container } = render(
    <Shop productsArr={productsArr} sortType="priceAsc" />
  );

  let productPrices = [...container.querySelectorAll('.priceDesc')];
  productPrices = productPrices.map((product) => product.textContent);
  let filteredProductPrices = productPrices.filter((a, b) => (a < b ? 1 : -1));

  expect(productPrices).toEqual(filteredProductPrices);
});

it('sorts products from highest to lowest review score', () => {
  let { container } = render(
    <Shop productsArr={productsArr} sortType="rating" />
  );

  let productRatings = [...container.querySelectorAll('[reviewavg]')];
  productRatings = productRatings.map((product) =>
    product.getAttribute('reviewavg')
  );
  let filteredProductRatings = productRatings.filter((a, b) =>
    a > b ? 1 : -1
  );

  expect(productRatings).toEqual(filteredProductRatings);
});
