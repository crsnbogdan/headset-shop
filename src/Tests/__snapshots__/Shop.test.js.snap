// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders corrently and matches snapshot 1`] = `
<div
  className="shop bg-blue-300"
>
  <div
    className="page__navoverlay"
  />
  <div
    className="shop__hero relative flex items-center justify-center bg-slate-200"
  >
    <h1
      className="text-white text-6xl font-bold"
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(-20px) translateZ(0)",
        }
      }
    >
      <div
        className="Typewriter"
        data-testid="typewriter-wrapper"
      />
    </h1>
  </div>
  <div
    className="shop__maincontainer py-12 px-24  bg-slate-100 h-full"
  >
    <div
      className="flex justify-between items-center mb-4"
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(-10px) translateZ(0)",
        }
      }
    >
      <h2
        className="text-2xl"
      >
        Products:
      </h2>
      <div
        className="MuiFormControl-root css-1nxucbq-MuiFormControl-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={true}
          id="demo-simple-select-label"
        >
          Sort by
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          onClick={[Function]}
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Review count
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value="reviews"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-14lo706"
            >
              <span>
                sortType
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="shop__productscontainer"
    >
      <div
        className="shop__productcontainer"
        style={Object {}}
      >
        <div
          className="shop__product flex flex-col justify-between"
          style={
            Object {
              "opacity": 0,
              "transform": "translateY(-20px) translateZ(0)",
            }
          }
        >
          <div
            className="shop__productimg"
            style={
              Object {
                "backgroundImage": "url(https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/5053/5053501_sd.jpg)",
              }
            }
          />
          <p
            className="font-semibold my-4"
          >
            Product 1(...)
          </p>
          <div
            className="flex mb-4"
            reviewAvg={4.7}
          >
            <span
              className="product__ratingcount text-sm mt-1 text-slate-400"
            >
              (2267)
            </span>
          </div>
          <p
            className="product__price product__price--sale font-semibold mb-4"
          >
            89.99
             $
          </p>
          <button
            className="product__addbtn overflow relative h-8 rounded-md w-full"
            onClick={[Function]}
            style={Object {}}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium addbtn__icon css-i4bv87-MuiSvgIcon-root"
              data-testid="ShoppingCartIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
             Add to Cart
          </button>
        </div>
      </div>
      <div
        className="shop__productcontainer"
        style={Object {}}
      >
        <div
          className="shop__product flex flex-col justify-between"
          style={
            Object {
              "opacity": 0,
              "transform": "translateY(-20px) translateZ(0)",
            }
          }
        >
          <div
            className="shop__productimg"
            style={
              Object {
                "backgroundImage": "url(https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/6298/6298657_sd.jpg)",
              }
            }
          />
          <p
            className="font-semibold my-4"
          >
            Product 2(...)
          </p>
          <div
            className="flex mb-4"
            reviewAvg={4.6}
          >
            <span
              className="product__ratingcount text-sm mt-1 text-slate-400"
            >
              (1437)
            </span>
          </div>
          <p
            className="product__price product__price--sale font-semibold mb-4"
          >
            99.99
             $
          </p>
          <button
            className="product__addbtn overflow relative h-8 rounded-md w-full"
            onClick={[Function]}
            style={Object {}}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium addbtn__icon css-i4bv87-MuiSvgIcon-root"
              data-testid="ShoppingCartIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
             Add to Cart
          </button>
        </div>
      </div>
      <div
        className="shop__productcontainer"
        style={Object {}}
      >
        <div
          className="shop__product flex flex-col justify-between"
          style={
            Object {
              "opacity": 0,
              "transform": "translateY(-20px) translateZ(0)",
            }
          }
        >
          <div
            className="shop__productimg"
            style={
              Object {
                "backgroundImage": "url(https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/6298/6298657_sd.jpg)",
              }
            }
          />
          <p
            className="font-semibold my-4"
          >
            Product 3(...)
          </p>
          <div
            className="flex mb-4"
            reviewAvg={3.7}
          >
            <span
              className="product__ratingcount text-sm mt-1 text-slate-400"
            >
              (100)
            </span>
          </div>
          <p
            className="product__price product__price--sale font-semibold mb-4"
          >
            79.99
             $
          </p>
          <button
            className="product__addbtn overflow relative h-8 rounded-md w-full"
            onClick={[Function]}
            style={Object {}}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium addbtn__icon css-i4bv87-MuiSvgIcon-root"
              data-testid="ShoppingCartIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
             Add to Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
